### ISSUE TITLE
mask_moe中T的计算似乎存在问题

### ISSUE THREAD

**Issue by vaeaea** [TROUBADOUR000/TimeFilter#8]
<img width="1675" height="727" alt="Image" src="https://github.com/user-attachments/assets/3e4b4eb0-2586-47ff-b3d3-f618fd20981e" />
在mask_moe中，计算S\T\ST时，我注意到S和ST都没有包含点k自身，但是T中包含了k自身，
并且后续mask的计算中有额外加上mask_base，因此是不是在计算T时也应该去除掉k自身的权重？
即修改为：

for k in range(L):
    S = ((torch.arange(L) % N == k % N) & (torch.arange(L) != k)).to(dtype).to(device) # 和当前patch,属于同一个时间步的所有变量（不含自身）
    T = ((torch.arange(L) >= k // N * N) & (torch.arange(L) < k // N * N + N))& (torch.arange(L) != k)).to(dtype).to(device) # 和当前patch，属于同一个变量的所有时间步
    T_ = ((torch.arange(L) >= k // N * N) & (torch.arange(L) < k // N * N + N)).to(dtype).to(device) # 和当前patch，属于同一个变量的所有时间步
    ST = torch.ones(L).to(dtype).to(device) - S - T_  # 和当前patch，不属于同一个时间步、不属于同一个变量的所有其余patch
    masks.append(torch.stack([S, T, ST], dim=0))

**Comment by TROUBADOUR000** [comment]
感谢你的细致观察！你的理解是正确的：在当前实现中，T 确实包含了点 k 自身，而 S 和 ST 都不包含，因此在后续加上 mask_base 时会造成不一致。

按照你的推理，在计算 T 时也应该去除自身的权重，这是合理且一致的处理方式。我们会据此更新代码，确保 S / T / ST 的定义完全对齐。

非常感谢你的反馈，也欢迎继续提出宝贵建议！
